Core/HLE/sceKernelInterrupt.cpp

```diff
@@ -826,6 +826,44 @@ static u32 sysclib_strncpy(u32 dest, u32 src, u32 size) {
 	return hleLogSuccessX(SCEKERNEL, dest);
 }
 
+static u32 sysclib_strtol(u32 strPtr, u32 endPtrPtr, int base) {	
+	const char* str;
+	str = Memory::GetCharPointer(strPtr);	
+	char* end = nullptr;
+	int result = strtol(str, &end, base);
+	ERROR_LOG(SCEKERNEL, "Untest %d = sysclib_strtol(%c, %08x, %i)",result, str, endPtrPtr, base);
+	if (Memory::IsValidRange(endPtrPtr, 4))
+		Memory::WriteUnchecked_U32(strPtr + (end - str), endPtrPtr);
+	return result;
+}
+
+static u32 sysclib_strchr(u32 src, int c) {
+	const std::string str = Memory::GetCharPointer(src);
+	ERROR_LOG(SCEKERNEL, "Untest sysclib_strchr(%c, %i)", str.c_str(),c);
+	int index = str.find(str,c);
+	if (index < 0) {
+		return 0;
+	}
+	return src + index;
+}
+
+static u32 sysclib_strrchr(u32 src, int c) {
+	const std::string str = Memory::GetCharPointer(src);
+	ERROR_LOG(SCEKERNEL, "Untest sysclib_strchr(%c, %i)", str.c_str(), c);
+	int index = str.rfind(str, c);
```

Comment:
This also wrong

Reply:
This also seems more or less right, `strrchr()` finds the pointer to the last occurrence of a character.  For example, if "hello" was at the pointer 0x1000, then `sysclib_strrchr(0x1000, 'l')` should return 0x1003.  `sysclib_strchr()` would return 0x1002 since it finds the first one.

-[Unknown]